msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java

```diff
@@ -287,16 +310,57 @@ public static AcquireTokenSilentOperationParameters createAcquireTokenSilentOper
                     requestAccountRecord.setLocalAccountId(multiTenantAccount.getId());
                 } else {
                     // Use that tenant's profile...
-                    final ITenantProfile tenantProfile = multiTenantAccount.getTenantProfiles().get(tenantId);
-
-                    validateClaimsExistForTenant(tenantId, tenantProfile.getClaims());
-
-                    requestAccountRecord.setUsername(
-                            SchemaUtil.getDisplayableId(tenantProfile.getClaims())
-                    );
-                    requestAccountRecord.setLocalAccountId(tenantProfile.getId());
+                    final ITenantProfile[] tenantProfile = {multiTenantAccount.getTenantProfiles().get(tenantId)};
```

Comment:
Quality doesn't hit the bar here.

- Do not block UI thread
- Guidance should say "prefer using tenantId authority"